packages/app-project/src/screens/ClassifyPage/components/ClassifierWrapper/ClassifierWrapper.js

```diff
@@ -71,26 +75,21 @@ export default function ClassifierWrapper({
     If the page URL contains a subject ID, update that ID when the classification subject changes.
     Subject page URLs can be either `/classify/workflow/{workflowID}/subject/{subjectID}`
     or `/classify/workflow/{workflowID}/subject-set/{subjectSetID}/subject/{subjectID}`.
+    Example: Subject Set Progress Banner arrow buttons
   */
-  const replaceRoute = router.replace
   let baseURL = `/${ownerSlug}/${projectSlug}/classify/workflow/${workflowID}`
   if (subjectSetID) {
     baseURL = `${baseURL}/subject-set/${subjectSetID}`
   }
-  let subjectPageURL = null
-  let subjectRouteChanged = false
-  if (subjectID) {
-    subjectPageURL = `${baseURL}/subject/${classifierSubjectID}`
-    subjectRouteChanged = router?.query.subjectID !== classifierSubjectID
-  }
 
-  useEffect(function updatePageURL() {
-    if (subjectPageURL && subjectRouteChanged) {
+  if (router.query.subjectID && classifierSubjectID) {
+    const subjectPageURL = `${baseURL}/subject/${classifierSubjectID}`
+    /* True when the classifier subject changed, but new id is not yet reflected in the url */
+    if (router?.query.subjectID !== classifierSubjectID) {
       const href = addQueryParams(subjectPageURL)
-      const as = href
-      replaceRoute(href, as, { shallow: true })
+      router.push(href, href, { shallow: true })
```

Comment:
These lines are checked and run on every render. They should be inside the `onSubjectChange` event handler, so that they only run once when the subject changes. That would also remove the need to track the classifier's internal subject state in this component.

Reply:
Something like this. I’m not sure why I replaced this with an effect hook. 

https://github.com/zooniverse/front-end-monorepo/blob/779759d04539cca83baa28c536d32eaaf0f415ba/packages/app-project/src/screens/ClassifyPage/components/ClassifierWrapper/ClassifierWrapper.js#L64-L72